### PR TITLE
Add address validation documentation

### PR DESCRIPTION
**Pull request by spokenbird**
[#184576997]

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1227,6 +1236,75 @@ address validation to work.
...
+whether Smarty validation for that address should be performed. The before mentioned `validate`
+parameter
+of the address fragment is used to determine the value of this hidden field.
+
+### Viewing the validated address
+
+We have provided a live template `cfa:addressValidationScreen` which will display a radio input with
+selections for the validated address that came back from Smarty if one was found, and the original
+address the user entered for them to select between. When using this live template you will be asked
+to enter values for the screen title, content replacement placeholder (this is a Thymeleaf variable
```
Comment (bseeger) [inline README.md:1302]: The parameters might be better defined in a list format here.  Might be easier to read.

File: README.md [hunk 2]
```diff
@@ -1227,6 +1236,75 @@ address validation to work.
...
+We have provided a `cfa:address` live template which provides form fields for Street Address, Apt
+Number,
+City, State and Zipcode. The address fragment in the live template has two parameters, `validate`
+and
+`inputName`. The `validate` parameter is a boolean (true or false) that determines whether the
+address will be
+validated by Smarty. The `inputName` parameter is the name of the associated address that will be
+linked
+to each of the Street Address, Apartment Number (as Street Address 2), City, State and Zipcode
+fields. In example, a inputName
```
Comment (bseeger) [inline README.md:1250]: Maybe "In this example,..."

File: README.md [hunk 3]
```diff
@@ -1227,6 +1236,75 @@ address validation to work.
...
+  "homeAddressState_validated": "CA",
+  "homeAddressZipCode_validated": "94710-1738"
+}
+```
+
+Note that both the original address and the validated address are stored, with the validated fields
+having the suffix `_validated`.
+
+Also note the input prefixed with `_validate` which appears here as `validatehomeAddress` with a
+value of true.
```
Comment (bseeger) [inline README.md:1291]: I think you're missing an underscore in the `validatehomeAddress`

**Comment by bseeger** [comment]
This looks good, overall!  I created a ticket for defining the screen level fragments, too:  https://www.pivotaltracker.com/story/show/184628800

You can leave the documenting of the address screen for that one, or feel free to add the address screen documentation as part of this PR.

File: README.md (hunk 1 above)
Comment (coltborg) [reply README.md:1302]: A table mayhaps?


 Parameter           | Default | Type                              | Description                                                       |
| ------------------- | ------- | --------------------------------- | ----------------------------------------------------------------- |
| title               | n/a     | String                            | A title for the whole screen.                                     |
| content             | n/a     | Thymeleaf identifier              | An identifier that matches the `ref`.                             |
| ref                 | n/a     | String                            | A unique identifier for the form content. Should match `content`. |
| addressInputToCheck | n/a     | Name of the input being verified. |                                                                   |
| inputName           | n/a     | String                            | Input name of the radio input that will show on screen.           |
| editAddressURL      | n/a     | String                            | Location to go back to edit the previously entered address.       |

Comment (spokenbird) [reply README.md:1302]: Went with the table :)

File: README.md (hunk 3 above)
Comment (spokenbird) [reply README.md:1291]: Thanks!

File: README.md [hunk 4]
```diff
@@ -1014,6 +1016,40 @@ They are visually displayed as three separate inputs for Month, Day and Year in
...
+your flow inputs class:
+
+```java
+String mailingAddressStreetAddress1;
+    String mailingAddressStreetAddress2;
+    String mailingAddressCity;
+    String mailingAddressState;
+    String mailingAddressZipCode;
+```
+
```
Comment (bseeger) [inline README.md:1052]: Most other sections have these words:  `For convenience, we have provided a XYZ live template` - can you add something to that affect here?

File: README.md [hunk 5]
```diff
@@ -1014,6 +1016,40 @@ They are visually displayed as three separate inputs for Month, Day and Year in
...
+Please note that when using the address fragment you will need to create corresponding fields in
+your
+flow inputs class for each of the above-mentioned inputs created by the fragment. For example, if
+your
+address fragments input name is mailingAddress, then you will need to create the following fields in
+your flow inputs class:
+
+```java
+String mailingAddressStreetAddress1;
```
Comment (bseeger) [inline README.md:1046]: Can you indent this to line. up with the other lines?

**Pull request merged**